# Consider all integer combinations of $a^{b}$ for $2 ≤ a ≤ 5$ and $2 ≤ b ≤ 5$:

# If they are then placed in numerical order, with any repeats removed, we get the following sequence of 15 distinct terms:

# 4, 8, 9, 16, 25, 27, 32, 64, 81, 125, 243, 256, 625, 1024, 3125

# How many distinct terms are in the sequence generated by $a^{b}$ for $2 ≤ a ≤ 100$ and $2 ≤ b ≤ 100$?

____

In [17]:
a = range(2,101)

In [18]:
set_values = set()

for b in range(2,101):
    values = set([x**b for x in a])
    set_values = set_values.union(values)

In [20]:
len(set_values)

9183

# Answer: 9183

____

### Note: I tried using numpy and taking `a**b` but some of the values were negative